doc/man3/SSL_CTX_set_early_cb.pod

```diff
@@ -0,0 +1,94 @@
+=pod
+
+=head1 NAME
+
+SSL_CTX_set_early_cb, SSL_EARLY_isv2, SSL_EARLY_get0_version, SSL_EARLY_get0_random, SSL_EARLY_get0_session, SSL_EARLY_get0_ciphers, SSL_EARLY_get0_compression_methods, SSL_EARLY_get0_ext - callback functions for early server-side ClientHello processing
+
+=head1 SYNOPSIS
+
+ typedef int (*ssl_early_cb_fn) (SSL *s, SSL_EARLY *ectx,
+                                 const unsigned char *clienthello, size_t ch_len,
+                                 int *al, void *arg);
+ void SSL_CTX_set_early_cb(SSL_CTX *c, ssl_early_cb_fn cb, void *arg);
+ int SSL_EARLY_isv2(SSL *s, SSL_EARLY *ectx);
```

Comment:
I thought about that, but the temptation to make SSL_EARLY just be reusing an existing type (CLIENTHELLO_MSG) was too strong, at least for the first crack.  The callback is almost certainly going to need the SSL object itself, though, so it seems reasonable to have it be a top-level API parameter.  I guess we could go the other way and attach the CLIENTHELLO_MSG to the SSL before invoking the callback, though.